chapter08/spg/exercise2.pl

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+my $file_handles = {};
+
+open my $input_file, '<', 'datafile.txt' or die "Could not open datafile.txt: $!";
+
+while ( <$input_file> ) {
+  my $line = readline $input_file;

```

Comment:
Ah, I based this on an example earlier in the chapter, but I guess that was wrong or I misinterpreted it.